.github/workflows/pullrequest.yml

```diff
@@ -56,3 +56,35 @@ jobs:
       - run: cargo fmt --all -- --check
       - run: cargo doc --all --exclude plexi-cli --all-features --document-private-items
       - run: cargo test --all --verbose --exclude plexi-cli --all-features
+
+  deploy-rust:
+    name: Deploy Rust Crates
+    timeout-minutes: 5
+    strategy:
+      matrix:
+        os: [ubuntu-24.04]
+    runs-on: ${{ matrix.os }}
+    if: ${{ github.ref == 'refs/heads/main' }}
+    needs:
+      - "test-rust"
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 1
+      - name: Cache
+        uses: actions/cache@v4
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            target
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+      - name: Set rust toolchain
+        run: rustup override set 1.87
+      - name: cargo login
+        run: |-
+          echo "${{ secrets.CRATES_IO_API_TOKEN }}" | cargo login
+      - run: cargo publish -p web-bot-auth  # will fail if we don't bump the version
+        continue-on-error: true
+      - run: cargo publish -p http-signature-directory # will fail if we don't bump the version
+        continue-on-error: true
```

Comment:
I think you should not echo the token. Use

`CARGO_REGISTRY_TOKEN` environemnt variable instead.

this would look as follow

```
env:
          CARGO_REGISTRY_TOKEN: ${{ secrets.CRATES_IO_API_TOKEN }}
```